pkg/tools/callback/readme.md

```diff
@@ -0,0 +1,67 @@
+## Why?
+
+NSM need a server to call client via GRPC, for this purpuse in native way it required to 
+have client serve a GRPC server on unix socket or TCP. 
+But to simplify communication between client and server a better way is to have 
+only one connection between client and server.
+
+And this grpc.callback library is intended to perform required functionality using 
+grpc bi-directional streams.
+
+## How?
+
+A Server declared as usual except:
+
+```go
+server := grpc.NewServer()
+...
+// register normal APIs on server
+...
+// Register callback GRPC endpoint and callback.Server will hold callback clients to be callable.
+callbackServer := callback.NewServer(callback.IdentityByAuthority)
+//
+callback.RegisterCallbackServiceServer(server, callbackServer)
+...
+_ = server.Serve(listener)
+```
+
+And then require to call client.
+```go
+target := "callback:{client-authority}"
+// If target is not in a list of callback server targets, 
+// it will perform grpc.DialContext to connect to passed target
+nsmClientGRPC, err := grpc.DialContext(context.Background(), target, callback.WithCallbackDialer(server.callbackServer, target), grpc.WithInsecure())
```

Comment:
Does this require grpc.WithInsecure()?

Reply:
Or to put it differently... can I use grpc.Peer from within the callback client?  It would be quite handy for the client using the callback to be able to get the correct AuthInfo for its peer :)

Reply:
Without insecure, it will be grpc error showing it require authentication, SSL could be used for this connection as well, but since it is passed using already authenticated connection I suppose better to not use it. It require individual grpc server on client to handle callbacks from particular server, so not sure if additional authentication or identification is required. Potentially we could enhance solution and have just one grpc for callbacks on client and connections to different servers, but current implementation doesn't support so.

Reply:
We will need to be able to get peer information for clients to properly compute tokens... is there any way to have this properly *set* the peer information?

Reply:
I've only found a way to retrieve client certificate.

Reply:
https://github.com/networkservicemesh/sdk/pull/171/files#diff-bcd6aa47c15ba3f1dd29728755ef360bR51

Reply:
OK... looking at the code more closely... this will in fact correctly handle the case of using the TLS in TLS if need be... or some simpler workarounds for Token Generation.